test/ng/httpSpec.js

```diff
@@ -1380,8 +1380,28 @@ describe('$http', function() {
             expect(errCallback.calls.mostRecent().args[0]).toEqualMinErr('$http', 'baddata');
           });
 
-        });
+          it('should not throw an error if JSON is invalid but content-type is not application/json', function() {
+            var errCallback = jasmine.createSpy('error');
+            $httpBackend.expect('GET', '/url').respond('{abcd}', {'Content-Type': 'text/plain'});
+
+            $http.get('/url').then(callback).catch(errCallback);
+            $httpBackend.flush();
+
+            expect(callback).toHaveBeenCalledOnce();
+            expect(errCallback).not.toHaveBeenCalled();
+          });
 
+          it('should return response unprocessed if JSON is invalid but content-type is not application/json', function() {
+            var response = '{abcd}';
+            $httpBackend.expect('GET', '/url').respond(response, {'Content-Type': 'text/plain'});
+
+            $http.get('/url').then(callback);
+            $httpBackend.flush();
+
+            expect(callback.calls.mostRecent().args[0].data).toMatch(response);
```

Comment:
`toBe()` looks for reference equality, `toMatch()` is responsible for matching strings and regexs. (according to jasmine documentation)